src/validate.py

```diff
@@ -396,6 +395,10 @@ def loop(validation_args_file: str, task_id: str = None, auto_clean_cache: bool
                 break  # Break the loop if no exception
             except KeyboardInterrupt:
                 break
+            except RuntimeError as e:
```

Comment:
You won't be able to capture this at the `loop` level, since it has already been caught in the `validate` function. You should handle this in the `validate` and then raise the exception to `loop`